File: jupyter2/ps/build_system/cat/jl/cat/cat.jl
## jl/cat/cat.jl
function main()
    # get command line arg
    filename = ARGS[1]
    # read file
    file = open(filename, "r")
    # get the whole contents of the file
    contents = read(file, String)
    # print the contents, without adding a newline at the end
    print(contents)
    # close the file
    close(file)
end

main()

File: jupyter2/ps/build_system/cat/go/cat/cat.go
/* go/cat/cat.go */
package main
import (
	"fmt"
	"os"
)

func main() {
	// get command line arguments
	filename := os.Args[1]
	// read file
	content, err := os.ReadFile(filename)
	if err != nil {
		fmt.Println("Error reading file:", err)
		return
	}
	// print content to stdout
	fmt.Print(string(content))
}
